snap/hooks/install

```diff
@@ -2,32 +2,33 @@
 
 set -eux
 
-# Create the config file that is used by pbm
-PBM_CONFIG_FILE="${SNAP_COMMON}/pbm_config.yaml"
+# Create the necessary parent directories 
+mkdir -p "${SNAP_DATA}/etc/pbm/"
+mkdir -p "${SNAP_DATA}/etc/mongod/"
+mkdir -p "${SNAP_COMMON}/var/lib/mongod/db"
+mkdir -p "${SNAP_COMMON}/var/log/mongod/"
+
+# Create the config file that is used by pbm along with the directories
+PBM_CONFIG_FILE="${SNAP_DATA}/etc/pbm/pbm_config.yaml"
 touch "${PBM_CONFIG_FILE}"
 
-# the pbm-agent daemon is not run as root meaning it does not have access write to SNAP_DATA, so
-# we need to create a child dir and give it access.
-mkdir -p "${SNAP_DATA}/pbm-agent-dir"
-
 # Copy over the mongod.conf and create needed directories/files
-MONGO_CONFIG_FILE="${SNAP_COMMON}/mongod.conf"
+MONGO_CONFIG_FILE="${SNAP_DATA}/etc/mongod/mongod.conf"
 echo "configuration file does not exist."
 echo "copying default config to ${MONGO_CONFIG_FILE}"
-cp -r ${SNAP}/etc/mongod.conf ${SNAP_COMMON}
+cp -r ${SNAP}/etc/mongod.conf ${MONGO_CONFIG_FILE}
 
 # Creating various files/directories in snap folders
 # This is required as the snap_daemon user doesnt have permission
 # to write to those directories
-mkdir -p ${SNAP_COMMON}/db
-touch ${SNAP_DATA}/mongod.log
-touch ${SNAP_DATA}/mongod.pid
-
-
-sed -i "s/fork: true/fork: false/g" $SNAP_COMMON/mongod.conf
-sed -i "s:/var/log/mongodb/mongod.log:$SNAP_DATA/mongod.log:g" $SNAP_COMMON/mongod.conf
-sed -i "s:/var/lib/mongodb:$SNAP_COMMON/db:g" $SNAP_COMMON/mongod.conf
-sed -i "s:/var/run/mongod.pid:$SNAP_DATA/mongod.pid:g" $SNAP_COMMON/mongod.conf
+touch ${SNAP_COMMON}/var/log/mongod/mongod.log
```

Comment:
I believe we should give privileges for writing into `${SNAP_COMMON}/var/{lib,log}/<component>` dirs for 584788:root instead of pre-creation of files